### PR TITLE
Clean up resources on exit and  on SIGINT/SIGTERM for consistency with java impl

### PR DESCRIPTION
**Pull request by minhbsq**
This PR fixes the root cause of the inability to restart node cleanly. 
TLDR: it is due to `aeron-go` not closing any resources (MediaDriver context, markfile, publications...). We have put aeron-go back on par with `aeron-java` in term of stability in this area.

- Before: Whenever we kill a node (service + consensus), if we restart it in less than **10 seconds**, the election is never concluded. Some nodes will stuck in state < 17 (terminal state for election)
- After: Same setup, restart after **1 second**. I tried 20 times and it always get to terminal state
Thus I think we can safely say that we have sufficiently solved the restart issue.

Next step:
- On all the code path that the service can terminate, we have to close resources. Right now, the only part left unhandled is panics. This will need individual service to handle resource closure in such cases.
Without this, we are still able to recover. Just need to restart all MediaDriver/Archive/Consensus/Services.

Original slack thread https://app.slack.com/client/T040R9VNPPS/C0504ARTY58

### CODE REVIEW

File: archive/archive.go [hunk 1]
```diff
@@ -628,7 +628,7 @@ func (archive *Archive) AddRecordedPublication(channel string, stream int32) (*a
 
 	archive.mtx.Lock()
 	defer archive.mtx.Unlock()
-	if err := archive.Proxy.StartRecordingRequest(correlationID, stream, true, true, sessionChannel); err != nil {
+	if err := archive.Proxy.StartRecordingRequest(correlationID, stream, true, false, sessionChannel); err != nil {
```
Comment (Aaronong) [inline archive/archive.go]: btw, did we check that this `true/false` thing is accurate?

File: cluster/clustered_service_agent.go [hunk 2]
```diff
@@ -169,15 +173,33 @@ func NewClusteredServiceAgent(
...
+			select {
+			case <-agent.signalChan:
+				// signal.Stop(agent.signalChan)
+				// close(agent.signalChan)
+				agent.OnClose()
+				return
+			default:
+				runtime.Gosched()
+			}
+		}
```
Comment (Aaronong) [inline cluster/clustered_service_agent.go]: Isn't this equivalent to 

```go
<-agent.signalChan
agent.OnClose()
```

File: archive/archive.go (hunk 1 above)
Comment (minhbsq) [reply archive/archive.go]: Let's not include this change into this PR, I'll split it to make the commit log cleaner

File: cluster/clustered_service_agent.go (hunk 2 above)
Comment (minhbsq) [reply cluster/clustered_service_agent.go]: Yes, indeed this is better
